### DISCUSSION TITLE
Image cropping

### DISCUSSION THREAD

**Discussion by louibel** [ImageMagick/ImageMagick6#323] [General]
### ImageMagick version

6.9.1-0 Q16

### Operating system, version and so on

OSX

### Description

Hello,

I'd like to cut an image like this :
- I define a position (Xc,Yc) in the image and the size of a tile (tWidth, tHeight).
- The first tile will therefore run from (Xc-tWidth/2 to Xc+tWidth/2) and (Yc-tHeight/2 and Yc-tHeight/2).
- See example in attached image where the center is represented by a circle in red and tiles border in black
![flower](https://github.com/ImageMagick/ImageMagick/assets/170869088/687f81c9-a339-4570-9f06-c1da872f27cd)

For the example, I've numbered the tiles from 1 to 9.

I've read page [https://www.imagemagick.org/Usage/crop/#crop_tile](url) but I can't really find an example that fits what I want and/or I didn't understand. Do you think it's possible?

If you could help me ...
Thanks

**Comment by snibgo** [comment]
I suggest `-roll` to put the top-left of the "t5" tile at the origin. Then `-crop tWidthxtHeight`. You might, or might not, then want `+repage`.

**Comment by snibgo** [comment]
Another possibility: `flip` and `-flop`, then `-crop tWidthxtHeight +repage`, then `flip` and `-flop` again. This doesn't depend on Xc and Yc.

I'm not sure exactly what you want.

**Comment by GeeMack** [comment]
With ImageMagick you can get a crop to start at any particular location by setting the page geometry before making the crop(s). This command will crop your sample image into the 12 separate pieces as outlined in green in this example...

    magick input.jpg -repage 950x750+150+150 -crop 300x300 result-%02d.png

Since your first tile isn't a whole 300x300, I added an extra 150 to the height and width geometry creating an offset to make sure the crop includes all the pieces.

![output](https://github.com/ImageMagick/ImageMagick/assets/65475279/8fcca753-0cbe-4a29-b559-d52076cce4fd)

That leaves those last pieces along the right as a narrow remainder from the edge. It will be a bit more complicated if you need to get those right-edge pieces to cut out on your black lines.

**Comment by fmw42** [comment]
I do not know if this will help, but you could try my overlapcrop Imagemagick script for Unix-like systems (Linux, Mac OSX, Windows Unix) at http://www.fmwconcepts.com/imagemagick/overlapcrop/index.php
